src/util.c

```diff
@@ -407,7 +407,7 @@ size_to_str(char *outbuf, size_t outbuflen, size_t size)
    }
 
    if (frac_val || is_approx) {
-      snprintf(outbuf, outbuflen, "~%ld.%ld %s", unit_val, frac_val, units);
+      snprintf(outbuf, outbuflen, "~%ld.%02ld %s", unit_val, frac_val, units);
```

Comment:
Here is the bug. It's a formatting issue. We want 2-digit scale, but old code will only print "7" rather than "07".

With fix, you will now see "07" when appropriate.